### PR TITLE
NAS documents general improvements (v2.8)

### PR DESCRIPTION
**Pull request by ultmaster**
### Description ###

Updates some docstrings and add more explanations.

#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [x] doc

### How to test ###

N/A

### CODE REVIEW

File: docs/source/nas/evaluator.rst [hunk 1]
```diff
@@ -42,6 +42,39 @@ The simplest way to customize a new evaluator is with :class:`FunctionalEvaluato
...
+
+The usage is shown below:
+
+.. code-block:: python
+
+   class Model(nn.Module):
+       ...
+
+   # Use a callable returning a model
+   evaluator.evaluate(Model)
```
Comment (QuanluZhang) [inline docs/source/nas/evaluator.rst:61]: this `Model` is a model space or a regular pytorch model?

File: docs/source/nas/evaluator.rst [hunk 2]
```diff
@@ -42,6 +42,39 @@ The simplest way to customize a new evaluator is with :class:`FunctionalEvaluato
...
+The underlying implementation of :meth:`~nni.retiarii.Evaluator.evaluate` depends on concrete evaluator that you used.
+For example, if :class:`~nni.retiarii.evaluator.FunctionalEvaluator` is used, it will run your customized fit function.
+If lightning evaluators like :class:`nni.retiarii.evaluator.pytorch.Classification` are used, it will invoke the ``trainer.fit()`` of Lightning.
+
+To evaluate an architecture that is exported from experiment (i.e., from :meth:`~nni.retiarii.experiment.pytorch.RetiariiExperiment.export_top_models`), use :func:`nni.retiarii.fixed_arch` to instantiate the exported model::
+
+    with fixed_arch(exported_model):
+        model = Model()
+    # Then use evaluator.evaluate
+    evaluator.evaluate(model)
```
Comment (QuanluZhang) [inline docs/source/nas/evaluator.rst:74]: do we need to add the `note` here to tell users that the same `evaluator` instance should not be used both for training supernet and for evaluating a subnet?

File: docs/source/nas/exploration_strategy.rst [hunk 3]
```diff
@@ -75,7 +75,14 @@ Starting from v2.8, the usage of one-shot strategies are much alike to multi-tri
 
    import nni.retiarii.strategy as strategy
    import nni.retiarii.evaluator.pytorch.lightning as pl
-   evaluator = pl.Classification(...)
+   evaluator = pl.Classification(
+     # Need to use pl.DataLoader instead of torch.utils.data.DataLoader here.
```
Comment (QuanluZhang) [inline docs/source/nas/exploration_strategy.rst:79]: might be better to addd one more sentence: users can refer to "serializer" doc if they stick to torch.utils.data.DataLoader

File: nni/retiarii/oneshot/pytorch/sampling.py [hunk 4]
```diff
@@ -76,6 +77,17 @@ def training_step(self, batch, batch_idx):
         self.resample()
         return self.model.training_step(batch, batch_idx)
 
+    def export(self) -> dict[str, Any]:
+        """
+        Export of Random one-shot. It will return an arbitrary architecture.
+        """
+        warnings.warn(
+            'Direct export from RandomOneShot returns an arbitrary architecture. '
+            'You might want to use the checkpoint in another search.',
```
Comment (QuanluZhang) [inline nni/retiarii/oneshot/pytorch/sampling.py:86]: -> Sampling the best architecture from this well trained supernet is another search process, users can use the supernet's checkpoint to do another search.

File: nni/retiarii/oneshot/pytorch/sampling.py [hunk 5]
```diff
@@ -105,22 +119,24 @@ class EnasLightningModule(RandomSamplingLightningModule):
...
     entropy_weight : float
-        Weight of sample entropy loss.
+        Weight of sample entropy loss in RL.
     skip_weight : float
-        Weight of skip penalty loss.
+        Weight of skip penalty loss. See :class:`~nni.retiarii.oneshot.pytorch.enas.ReinforceController` for details.
     baseline_decay : float
-        Decay factor of baseline. New baseline will be equal to ``baseline_decay * baseline_old + reward * (1 - baseline_decay)``.
+        Decay factor of reward baseline, which is used to normalize the reward in RL.
+        At each step, the new ew baseline will be equal to ``baseline_decay * baseline_old + reward * (1 - baseline_decay)``.
```
Comment (QuanluZhang) [inline nni/retiarii/oneshot/pytorch/sampling.py:129]: ew baseline?

File: docs/source/nas/evaluator.rst (hunk 2 above)
Comment (ultmaster) [reply docs/source/nas/evaluator.rst:74]: I think the problem is restricted to pytorch-lightning.

Comment (ultmaster) [reply docs/source/nas/evaluator.rst:76]: No. There is no such implication.

**Review by QuanluZhang — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**